sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -82,14 +83,30 @@ export abstract class SwapRouter {
     positionManagerOverride?: string
   ): MethodParameters {
     const v4Pool: V4Pool = options.outputPosition.pool
-    const token0 = options.inputPosition.pool.token0
-    const token1 = options.inputPosition.pool.token1
+    let token0: Currency = options.inputPosition.pool.token0
+    let token1: Currency = options.inputPosition.pool.token1
     const v4PositionManagerAddress =
       positionManagerOverride ?? CHAIN_TO_ADDRESSES_MAP[v4Pool.chainId as SupportedChainsType].v4PositionManagerAddress
 
+    const nativeCurrency = Ether.onChain(v4Pool.chainId)
+
+    // if migrating wrapped native -> native
+    if (nativeCurrency.wrapped.equals(token0) || nativeCurrency.wrapped.equals(token1)) {
```

Comment:
this is hard to read because its unclear that `token0` is of the v3 pool not v4. I think some renaming could help make that clearer that these tokens are v3

Reply:
also just because a v3 pool has WETH, doesnt mean that the user wants to migrate to ETH? this fi statement will enter for WETH -> WETH migrations too wont it?

Reply:
ah okay i just asked if we are strictly preventing this case or not